### DISCUSSION TITLE
Warn users if re-render an existing doc/project with an older version of Quarto

### DISCUSSION THREAD

**Discussion by guslipkin** [quarto-dev/quarto-cli#5790] [Q&A] [answered]
### Description

A user is working across multiple computers and updates Quarto on one computer but forgets to do so on the other. They then begin using the second computer, and re-render an existing document or project, thereby downgrading the Quarto version.

I think this would be good to integrate into the CLI so that things don't break when new features are released.

I did my best to dig in and see if there is checking and I just haven't encountered it yet, but I didn't find anything related to `kQuartoVersion` that looked right.

**Comment by cscheid** [comment, marked as answer]
https://github.com/quarto-dev/quarto-cli/blob/main/src/command/render/render-files.ts#L108

It takes a semver-compatible string. It's meant to be used in projects, like so: https://quarto.org/docs/extensions/project-types.html#complete-example but it works in individual documents as well.

**Reply by guslipkin** [reply]
Thank you! I guess I was thinking at the wrong level, but it does make more sense to be a user optional/updated value.

In case anyone else finds this, here's a little auto-updater I made to leave in my `index.qmd` that will update the `quarto-version` in your yaml if you render with a higher version than what's already there.

Edit: `write_yaml` defaults to yes/no instead of true/false

```
#| echo: false
#| output: false
y <- yaml::read_yaml("_quarto.yml")
if (is.null(y[["quarto-version"]])) {
  y[["quarto-version"]] <- paste0('>=', quarto::quarto_version())
} else {
  cur_ver <-
    y[["quarto-version"]] |>
    strsplit("[\\.<>=]") |>
    unlist()
  cur_ver <- as.integer(cur_ver[cur_ver != ""])
  quarto_ver <- 
    quarto::quarto_version() |>
    as.character() |>
    strsplit("\\.") |>
    unlist() |>
    as.integer()
  if (all(quarto_ver >= cur_ver)) { 
    y[["quarto-version"]] <- paste0('>=', quarto::quarto_version())
  }
}
yaml::write_yaml(
  y, 
  "_quarto2.yml", 
  handlers = list(
    logical = \(x) {
      x <- ifelse(x, "true", "false")
      class(x) <- "verbatim"
      return(x)
    }
  )
)
```

**Reply by cscheid** [reply]
There's also the `<meta generator="">` tag in HTML formats which you could inspect. That one is always generated.
